src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -704,11 +705,14 @@ internal void Stop()
                     }
                 }
             }
+            return Task.CompletedTask;
         }
 
-        internal void Start()
+        private Task OnRuntimeInitializeStart(CancellationToken tc)
         {
+            typeManager.Start();
```

Comment:
May want a 'load resources' stage prior to runtime initialization stage for assembly loading and related.